src/picongpu/include/plugins/hdf5/restart/LoadParticleAttributesFromHDF5.hpp

```diff
@@ -76,7 +76,20 @@ struct LoadParticleAttributesFromHDF5
 
         const std::string name_lookup[] = {"x", "y", "z"};
 
-        ComponentType* tmpArray = new ComponentType[elements];
+        ComponentType* tmpArray = NULL;
+        if( elements > 0 )
+            tmpArray = new ComponentType[elements];
+        else
+        {
+            /** This is a workaround to avoid the bug that processes hangs
+             *  if some call `DataCollector::read()` where elements to read is
+             *  zero and the destination pointer for the read data is set to `NULL`.
+             *  see: - `PIconGPU`  issu: https://github.com/ComputationalRadiationPhysics/picongpu/pull/611
+             *       - `libSplash` issu: https://github.com/ComputationalRadiationPhysics/libSplash/issues/148
+             * \todo: please remove this workaround after the libsplash bug is fixed
+             */
+            tmpArray = new ComponentType[1];
```

Comment:
**Update:** no, the work-around is valid but we have an other bug with static load balancing restarts right now #637